app/jobs/payout/concerns/retries_from_last_enqueued_publisher.rb

```diff
@@ -0,0 +1,42 @@
+module RetriesFromLastEnqueuedPublisher
+  extend ActiveSupport::Concern
+
+  def enqueue_retryable_potential_payout!
+    # self.class.name is the name of the caller function
+    last_enqueued_publisher_key = "#{self.class.name}-#{@payout_report_id || ""}"
```

Comment:
Can you write up a section about what the technical underlying issue is and why this method fixes it?

Reply:
Discussed offline. Resolving